### PR TITLE
Update cost queries to match the new cost export structure 

### LINKED ISSUE #2992: District heating costs per household contain mistake in heat delivery system costs
The vertical_stacked_bar_ht_district_heating_costs_per_household figure (and mt and lt versions) contain the heat delivery system costs per household. These costs are calculated by taking the ht heat delivery system capital and depreciation costs (capex) and dividing those by the number of residences in the district heating network ([link](https://github.com/quintel/etsource/blob/2ec8296e803fe0258e7e0e148f3723f2211d3879/gqueries/output_elements/output_series/vertical_stacked_bar_ht_district_heating_costs_per_household/households_ht_heat_delivery_system_costs_per_household/households_ht_heat_delivery_system_capex_per_household.gql)):

```ruby
- unit = eur
- query =
    number_of_residences_in_ht_heating_network = V(households_space_heater_district_heating_ht_steam_hot_water,number_of_units);
    ht_heat_delivery_system_total_annual_costs = 
        SUM(
            Q(households_ht_heat_delivery_system_depreciation_costs),
            Q(households_ht_heat_delivery_system_capital_costs)
        );

    DIVIDE(
        ht_heat_delivery_system_total_annual_costs,
        number_of_residences_in_ht_heating_network
    )
```

However, the heat delivery system capex is based on ALL residences with that system, not just the residences that have district heating. As a result, the capex is too high, and therefore the costs per household as well.

This query should recalculate the capex for district heating households only to correct for this mistake.

### PR DESCRIPTION
**Pull request by redekok**
This PR renames and restructures costs queries to match the new structure of the costs export functionality. It also fixes minor bugs.

**Background**
The new costs export serializer finds queries with the following suffixes:

- `_annualised_costs`
- `_capex`
- `_opex`
- `_capital_costs`
- `_depreciation_costs`
- `_fixed_operation_and_maintenance_costs`
- `_variable_operation_and_maintenance_costs`
- `_investment_costs`
- `_wacc`
- `_technical_lifetime`

Queries within all infrastructure components (heat, electricity, gas, hydrogen) have been changed to match these suffixes.

**Changes**
_Heat infrastructure costs_

- add layer of `_capex` & `_opex` queries for all heat infrastructure components;
- add `_wacc` and `_technical_lifetime` queries;
- refer to `_wacc` and `_technical_lifetime` queries when calculating `_capital_costs` and `_depreciation_costs`;
- rename `_yearly_maintenance_costs` queries to `_fixed_operation_and_maintenance_costs`;
- rename `distribution_pipelines` queries to distinguish between **main** distribution pipelines and the overall distribution pipeline system (=main distribution pipelines + connection pipelines). See [documentation](https://docs.energytransitionmodel.com/main/heat-infrastructure-costs#Secondary-distribution-pipelines)
- move `_indoor_costs` queries to main_queries folder (for consistency with other infrastructure components); 

_Electricity infrastructure costs_
Existing query names were inconsistent with other costs queries. For example, `lv_net_additional_capital_costs` actually referred to additional CAPEX = capital costs + depreciation costs. 

These and other inconsistencies have been addressed:
- Restructure costs queries using `_costs` and `_present` / `_future` / `_delta_present_future` name system;
- add `_capital_costs` and `_depreciation_costs` queries;
- add `_wacc` and `_technical_lifetime` queries
- Rename `total_costs` queries to `annualised_costs`;
- rename `total_investment_costs` to `investment_costs`;
- rename `additional_operation_and_maintenance_costs` to `fixed_operation_and_maintenance_costs`;
- add layer of `_capex` & `_opex` queries;

_Gas infrastructure costs_
Add layer of `_annualised_costs` query

_Heat delivery system costs_

- Add layer of `_annualised_costs` query
- add `_wacc` and `_technical_lifetime` queries for all heat delivery systems;
- Recalculates costs for _all_ households and buildings that have ht/mt/lt heat delivery systems. Closes #2992 

_Carrier costs_
Add layer of `_annualised_costs` query

_Hydrogen storage and insulation costs_
Add layer of `_annualised_costs` and `_capex` queries

_Insulation costs buildings & households_

- Add layer of `_annualised_costs` query
- Calculate `_capex` by dividing insulation costs by technical lifetime
- Move queries into folder structure

_CO2 emissions costs_
Add separate `ccus` and `co2_emissions_costs` queries.

_Bug fixing_
- Rename `cost_co2` group to `costs_co2_ccus`
- Remove `costs_building_and_installations_transport` group from relevant nodes and costs figures;
- add buildings hybrid heatpumps to `costs_building_and_installations` group;
- Update takes_part_in_ets for agriculture and backup burners based on input capacity

### CODE REVIEW

**Comment by mabijkerk** [comment]
I'm having a hard time following the new structure for CCUS costs queries:

- costs_co2: costs _co2_energy_graph + costs_molecule_graph
- costs_co2_ccus: costs_co2_energy_graph_ccus + costs_molecule_graph
- costs_co2_energy_graph: costs_co2_energy_graph_ccus + costs_co2_energy_graph_co2_emissions_costs
- costs_co2_energy_graph_ccus: costs of CCUS CAPEX and OPEX of nodes in the energy graph
- costs_co2_energy_graph_co2_emissions_costs: costs of CO2 emissions of nodes in the energy graph
- costs_molecule_graph: costs of CAPEX and OPEX in the molecule graph **excluding CCS**

What I'm struggling with:

- The main group is called costs_co2 in the download, with ccus as a subgroup, but in the folder in ETSource the main group is 6_CCUS
- There is overlap between the queries costs_co2 and costs_co2_ccus and between costs_co2_ccus and costs_co2_energy_graph
- The costs_molecule_graph are not called costs_co2_... and are excluding CCS?
- The second "costs" in costs_co2_energy_graph_co2_emissions_costs feels redundant

Most importantly, based on the query titles I have no idea what's what. When is something "CO2", when is something "CCUS".

**Comment by mabijkerk** [comment]
I noted that the insulation costs do not have a capital costs component. @KoenvanB noted this as well and added that the source data may already have applied some kind of annualisation. This is something to note for further investigation.

**Comment by mabijkerk** [comment]
Another point to note is that our gas infrastructure costs 'calculation' is extremely limited. It does not even change based on changes in demand or supply of gas.

File: gqueries/general/costs/mece_costs/6_CCUS/costs_co2_energy_graph_ccus.gql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      V(G(costs_production_power_plants), operating_expenses_ccs_per(:node)),
+      V(G(costs_production_chp_plants), capital_expenditures_ccs_per(:node)),
+      V(G(costs_production_chp_plants), operating_expenses_ccs_per(:node)),
+      V(G(costs_production_dedicated_hydrogen_production), capital_expenditures_ccs_per(:node)),
+      V(G(costs_production_dedicated_hydrogen_production), operating_expenses_ccs_per(:node)),
+      V(G(costs_co2_ccus), capital_expenditures_excluding_ccs_per(:node)),
+      V(G(costs_co2_ccus), capital_expenditures_ccs_per(:node)),
+      V(G(costs_co2_ccus), operating_expenses_excluding_ccs_per(:node)),
+      V(G(costs_co2_ccus), operating_expenses_ccs_per(:node))
+    )
```
Comment (mabijkerk) [inline gqueries/general/costs/mece_costs/6_CCUS/costs_co2_energy_graph_ccus.gql:17]: @KoenvanB why wouldn't you include all the costs groups here? If someone then adds CCS in e.g. the households sector, the costs queries will automatically be updated properly.

**Comment by mabijkerk** [comment]
The queries for heat infrastructure are rather numerous, making it hard to review (and maintain) them. This is something to note for further improvements.

File: gqueries/general/costs/mece_costs/6_CCUS/costs_co2_energy_graph_ccus.gql (hunk 1 above)
Comment (KoenvanB) [reply gqueries/general/costs/mece_costs/6_CCUS/costs_co2_energy_graph_ccus.gql:17]: I agree. I have limited myself to the costs groups that were already here, but it is better to simply include all of them (like I did in the `co2_emissions_costs` query)

**Pull request merged**